File: src/com/l2kt/gameserver/skills/effects/EffectCombatPointHealOverTime.java
package com.l2kt.gameserver.skills.effects;

import com.l2kt.gameserver.model.L2Effect;
import com.l2kt.gameserver.skills.Env;
import com.l2kt.gameserver.network.serverpackets.StatusUpdate;
import com.l2kt.gameserver.templates.skills.L2EffectType;

class EffectCombatPointHealOverTime extends L2Effect
{
	public EffectCombatPointHealOverTime(Env env, EffectTemplate template)
	{
		super(env, template);
	}
	
	@Override
	public L2EffectType getEffectType()
	{
		return L2EffectType.COMBAT_POINT_HEAL_OVER_TIME;
	}
	
	@Override
	public boolean onActionTime()
	{
		if (getEffected().isDead())
			return false;
		
		double cp = getEffected().getCurrentCp();
		double maxcp = getEffected().getMaxCp();
		cp += calc();
		
		if (cp > maxcp)
			cp = maxcp;
		
		getEffected().setCurrentCp(cp);
		StatusUpdate sump = new StatusUpdate(getEffected());
		sump.addAttribute(StatusUpdate.CUR_CP, (int) cp);
		getEffected().sendPacket(sump);
		return true;
	}
}

File: src/com/l2kt/gameserver/skills/effects/EffectCombatPointHealOverTime.kt
package com.l2kt.gameserver.skills.effects

import com.l2kt.gameserver.model.L2Effect
import com.l2kt.gameserver.network.serverpackets.StatusUpdate
import com.l2kt.gameserver.skills.Env
import com.l2kt.gameserver.templates.skills.L2EffectType

internal class EffectCombatPointHealOverTime(env: Env, template: EffectTemplate) : L2Effect(env, template) {

    override fun getEffectType(): L2EffectType {
        return L2EffectType.COMBAT_POINT_HEAL_OVER_TIME
    }

    override fun onActionTime(): Boolean {
        if (effected.isDead())
            return false

        var cp = effected.currentCp
        val maxcp = effected.maxCp.toDouble()
        cp += calc()

        if (cp > maxcp)
            cp = maxcp

        effected.currentCp = cp
        val sump = StatusUpdate(effected)
        sump.addAttribute(StatusUpdate.CUR_CP, cp.toInt())
        effected.sendPacket(sump)
        return true
    }
}
